src/DotVVM.Framework/package.json

```diff
@@ -4,10 +4,17 @@
   "private": true,
   "devDependencies": {
     "uglify-js": "3.1.10",
-    "watch": "1.0.2"
+    "typescript": "^2.5.3",
+    "concurrently": "^3.5.1",
+    "onchange": "^3.3.0"
   },
   "scripts": {
-    "bundle": "npm run bundle:js",
-    "bundle:js": "uglifyjs -o Resources/Scripts/Polyfills/polyfill.bundle.js -c -m -- Resources/Scripts/Polyfills/promise.js Resources/Scripts/Polyfills/fetch.js"
+    "build": "npm run compile:ts && npm run minify:dotvvm:js && npm run bundle:polyfill:js",
+    "compile:ts": "tsc",
+    "minify:dotvvm:js": "uglifyjs -o Resources/Scripts/DotVVM.min.js -c -m -- Resources/Scripts/DotVVM.js",
+    "bundle:polyfill:js": "uglifyjs -o Resources/Scripts/Polyfills/polyfill.bundle.js -c -m -- Resources/Scripts/Polyfills/promise.js Resources/Scripts/Polyfills/fetch.js",
+    "watch": "concurrently --kill-others \"npm run watch:ts\" \"npm run watch:js\"",
```

Comment:
What is the advantage of using `concurrently`?

Reply:
`concurrently` runs multiple tasks at once which is needed because `watch:ts` task never ends and therefore it's not possible to use `&&`

Reply:
Have you tried `npm run watch:ts | npm run watch:js`?

Reply:
It works too, but the first task `stdout` is sent to the second task `stdin`. It might lead to unpredictable behavior in a future.